portfolio/src/main/webapp/script.js

```diff
@@ -60,12 +65,33 @@ function loadComments() {
 /** Creates a list item that contains the comment */
 function createCommentElement(comment){
 
+  // create list elements
   const commentElement = document.createElement('li');
-  const userCommentElement = document.createElement('span');
-  userCommentElement.innerText = comment.userComment;
+  const userCommentElement = document.createElement('div');
+  const userNameElement = document.createElement('div');
+  const commentSpacing = document.createElement('br');
+  
+  empty = false;
+  // deal with empty inputs
+  if(comment.userName == "" || comment.userName == undefined){
+      comment.userName = "Anonymous";
+  }
+  if(comment.userComment == ""){
+      empty = true;
+  } 
 
-  commentElement.appendChild(userCommentElement);
-  return commentElement;
+  if(!empty){
+    // populate list elements with name and comment
+    userCommentElement.innerText = comment.userComment;
+    userNameElement.innerText = ("-" + comment.userName);
+  
+    // add to list item
+    commentElement.appendChild(userCommentElement);
+    commentElement.appendChild(userNameElement);
+    commentElement.appendChild(commentSpacing);
+    return commentElement;
+  }
+  return null;
 }
```

Comment:
I don't understand javascript well enough to understand this function. Would you mind walking me through this in a call tomorrow, please?

Reply:
What was confusing me is whether `document.createElement()` has any side effects (such as storing the new element somewhere within `document`). Devon and I are discussing this. But I don't think it actually matters. More of a point of curiosity. (LMK if you know the answer.)